superset/views/core.py

```diff
@@ -1436,7 +1437,8 @@ def save_or_overwrite_slice(
         slc.viz_type = form_data['viz_type']
         slc.datasource_type = datasource_type
         slc.datasource_id = datasource_id
-        slc.slice_name = slice_name
+        slc.slice_name = slice_name or slc.slice_name
```

Comment:
I am quite sure session.merge(slc) takes care of merging with the existing entry. Remove the 'or' clause and see if this feature works as intended.

Reply:
It does, however if the user updates the description but not the title the query param returns `None` and this overwrites the chart name with `None`.